src/components/Form/fields.spec.jsx

```diff
@@ -2,37 +2,17 @@ import React from 'react'
 import { Field, Form } from 'react-final-form'
 import renderer from 'react-test-renderer'
 
-import { AdaptedInput, AdaptedTextarea } from './fields'
+import { DumbField } from './fields'
 
-describe('AdaptedInput', () => {
+describe('DumbField', () => {
   it('should adapt cozy-ui input for react-final-form', () => {
```

Comment:
I'm not sure what exactly the test was looking for (the problem with snapshots), but I suspect it may be related to the `data-valid` thing. I think we should update the test, depending on what exactly the component should do.

Reply:
En fait avant on avait un composant qui retourné soit un input soit un textarea. Et il y avait 2 tests, pour tester les 2 cas. Maintenant avec le Textfield il n'y a plus qu'un seul composant à tester. Le test, sauf erreur de ma part, n'utilisait pas les data-valid mais faisait juste un snapshot de rendu, pour vérifier qu'on retournait bien un input ou un textarea. Enfin de ce que j'ai compris...

Reply:
Je pense que non 😁. Le fichier exportait 2 composants, mais la logique de choix entre l'un et l'autre n'était pas dans ce fichier, et pas testé. A mon avis ca a vraiment un rapport avec ce `data-valid` qui est ajouté par l'ancien composant, et vu la description du test, je pense que `data-valid` est utilisé par react-final-form. En tout cas je pense que c'est important de comprendre a quoi il sert et si on peut vraiment l'enlever avant de voir ce qu'il est pertinent de tester.

Reply:
> > Je pense qu'on gagnerait à tout refacto au propre la prochaine fois qu'on travail sur ces champs. Je me suis dit que ça arrivera probablement, et qu'il fallait mieux sortir une PR en l'état (ce qui a permis une première prise en main et compréhension) et tout refacto ensuite.
> 
> Yep, pas de problème. En revanche la que c'est frais, ca vaux le coup de faire une issue dans ce dépot avec:
> 
>     * une explication de l'archi actuelle
> 
>     * pourquoi on veux la refacto
> 
>     * des pistes pour la nouvelle archi si tu en as

Ok je m'en occupe :)